lib/HLSL/HLMatrixLowerPass.cpp

```diff
@@ -537,6 +545,24 @@ void HLMatrixLowerPass::replaceAllVariableUses(
       continue;
     }
 
+    if (BitCastInst *BCI = dyn_cast<BitCastInst>(Use.getUser())) {
+      // Replace bitcasts to i8* for lifetime intrinsics.
+      if (BCI->getType()->isPointerTy() && BCI->getType()->getPointerElementType()->isIntegerTy(8))
+      {
+        DXASSERT(BCI->getNumUses() == 1, "expected only a single use");
```

Comment:
Couldn't one bitcast be used for both lifetime.start and lifetime.end in a single block?  Why can it only have one use?

Reply:
Absolutely, though currently there's always one bitcast per intrinsic. Makes sense to be more flexible here. Fixed.